TweetMate/src/main/java/com/ybsystem/tweetmate/usecases/StatusUseCase.java

```diff
@@ -0,0 +1,282 @@
+package com.ybsystem.tweetmate.usecases;
+
+import android.app.Activity;
+import android.net.Uri;
+
+import com.ybsystem.tweetmate.application.TweetMateApp;
+import com.ybsystem.tweetmate.libs.eventbus.PostEvent;
+import com.ybsystem.tweetmate.libs.eventbus.StatusEvent;
+import com.ybsystem.tweetmate.models.entities.twitter.TwitterStatus;
+import com.ybsystem.tweetmate.databases.PrefAppearance;
+import com.ybsystem.tweetmate.databases.PrefSystem;
+import com.ybsystem.tweetmate.utils.DialogUtils;
+import com.ybsystem.tweetmate.utils.ExceptionUtils;
+import com.ybsystem.tweetmate.utils.StorageUtils;
+import com.ybsystem.tweetmate.utils.ToastUtils;
+
+import org.greenrobot.eventbus.EventBus;
+
+import java.io.File;
+import java.util.List;
+
+import id.zelory.compressor.Compressor;
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.observers.DisposableObserver;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import twitter4j.StatusUpdate;
+import twitter4j.Twitter;
+import twitter4j.TwitterException;
+
+import static com.ybsystem.tweetmate.models.enums.ConfirmAction.*;
+import static com.ybsystem.tweetmate.resources.ResString.*;
+import static com.ybsystem.tweetmate.resources.ResString.STR_SUCCESS_UNLIKE;
+
+public class StatusUseCase {
+
+    private StatusUseCase() {
+    }
+
+    public static void retweet(TwitterStatus status) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS3776)